comp110-worksheet-6/OxoBoard.cs

```diff
@@ -8,41 +8,102 @@ namespace comp110_worksheet_6
+
+        // Constructor. Perform any necessary data initialisation here.
+        // Uncomment the optional parameters if attempting the stretch goal -- keep the default values to avoid breaking unit tests.
+        public OxoBoard(int width = 3, int height = 3)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            //initialize 3x3 2D char array
+            board = new Mark[3,3];
+            for(int x =0; x<width; x++)
+            {
+                for(int y=0;y<height; y++)
+                {
+                    board[x,y] = Mark.None;
+                }
+            }
 		}
 
 		// Return the contents of the specified square.
 		public Mark GetSquare(int x, int y)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            return board[x, y];
 		}
 
 		// If the specified square is currently empty, fill it with mark and return true.
 		// If the square is not empty, leave it as-is and return False.
 		public bool SetSquare(int x, int y, Mark mark)
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
+            //Check if x and y are within the bounds of the array.
+            if ((x>= 0 && y>=0) && (x <board.Length && y < board.Length))
+            {
+                if (board[x, y] == Mark.None)
+                {
+                    board[x, y] = mark;
+                    return true;
+                }
+            }
+            //Since the function will stop if it returns something, if we get to this point we can assume that the given parameters weren't valid or the place wasn't empty.
+            return false;
 		}
 
-		// If there are still empty squares on the board, return false.
-		// If there are no empty squares, return true.
+		// If there are still empty board on the board, return false.
+		// If there are no empty board, return true.
 		public bool IsBoardFull()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            for (int x = 0; x < 3; x++)
+            {
+                for (int y = 0; y < 3; y++)
+                {
+                    if (board[x, y] != Mark.None)
+                        continue;
+                    return false;
+                }
+            }
+        
+            return true;
+        }
 
-		// If a player has three in a row, return Mark.O or Mark.X depending on which player.
-		// Otherwise, return Mark.None.
-		public Mark GetWinner()
+        // If a player has three in a row, return Mark.O or Mark.X depending on which player.
+        // Otherwise, return Mark.None.
+        public Mark GetWinner()
 		{
-			throw new NotImplementedException("TODO: implement this function and then remove this exception");
-		}
+            //Check for 3 of the same in each row
+            for (var y = 0; y < board.GetLength(0); y++)
```

Comment:
It doesn't really matter as they're both the same, but shouldn't this be `board.GetLength(1)`?